src/main/java/com/jozufozu/flywheel/compat/CompatHelper.java

```diff
@@ -0,0 +1,34 @@
+package com.jozufozu.flywheel.compat;
+
+import java.util.function.Supplier;
+
+import com.google.common.base.Suppliers;
+
+import net.fabricmc.loader.api.FabricLoader;
+import net.fabricmc.loader.api.Version;
+import net.fabricmc.loader.api.metadata.version.VersionPredicate;
+import net.fabricmc.loader.impl.util.version.VersionPredicateParser;
+
+public class CompatHelper {
+	public static final Supplier<Boolean> IS_SODIUM_LOADED = Suppliers.memoize(() -> FabricLoader.getInstance().isModLoaded("sodium"));
+	public static final Supplier<Boolean> IS_EMBEDDIUM_LOADED = Suppliers.memoize(() -> FabricLoader.getInstance().isModLoaded("embeddium"));
+
+	public static boolean IS_SODIUM_0_5 = false;
+	public static boolean IS_SODIUM_0_6 = false;
+
+	public static void initSodiumVersion() {
+		if (IS_SODIUM_LOADED.get() && !IS_EMBEDDIUM_LOADED.get()) {
+			try {
+				VersionPredicate predicate0_5 = VersionPredicateParser.parse(">=0.5.0 <0.6.0");
+				VersionPredicate predicate0_6 = VersionPredicateParser.parse("<0.6 >=0.6.0-beta.2");
+				Version sodiumVersion = FabricLoader.getInstance()
+						.getModContainer("sodium")
+						.orElseThrow()
+						.getMetadata()
+						.getVersion();
+				IS_SODIUM_0_5 = predicate0_5.test(sodiumVersion);
+				IS_SODIUM_0_6 = predicate0_6.test(sodiumVersion);
+			} catch (Throwable ignored) {}
+		}
+	}
```

Comment:
`IS_SODIUM_*` should be final and the logic to compute the values of these fields should be in a static initializer block located right before these fields to prevent ordering issues. `initSodiumVersion` should be removed. Use `VersionPredicate.parse` instead of `VersionPredicateParser.parse` to avoid using implementation classes. Retrieve the Sodium version first, then continue with the logic only if the version was retrieved without an exception.